add-on/chrome/bootstrap.js

```diff
@@ -481,11 +481,64 @@ var WindowListener = {
         gBrowser.tabContainer.removeEventListener("TabSelect", this);
         gBrowser.removeEventListener("DOMTitleChanged", this);
         gBrowser.removeEventListener("mousemove", this);
+        this.removeRemoteCursor();
         this._listeningToTabSelect = false;
         this._browserSharePaused = false;
       },
 
       /**
+       *  If sharing is active, paints and positions the remote cursor
+       *  over the screen
+       *
+       *  @param cursorData Object with the correct position for the cursor
+       *                    {
+       *                      ratioX: position on the X axis (percentage value)
+       *                      ratioY: position on the Y axis (percentage value)
+       *                    }
+       */
+      addRemoteCursor: function(cursorData) {
+        if (!this._listeningToTabSelect) {
+          return;
+        }
+
+        let browser = gBrowser.selectedBrowser;
+        let cursor = browser.parentNode.querySelector("#remote-cursor");
+
+        if (!cursor) {
+          cursor = document.createElement("image");
+          cursor.setAttribute("id", "remote-cursor");
+          cursor.style.pointerEvents = "none";
+        }
+
+        // Calculate the final position based on browser's size.
+        let position = {
+          left: cursorData.ratioX * browser.boxObject.width,
+          top: cursorData.ratioY * browser.boxObject.height
+        };
```

Comment:
Might as well just calculate these inline the `setAttribute` instead of creating this object.